src/main/java/org/apache/zeppelin/interpreter/util/ProcessLauncher.java

```diff
@@ -0,0 +1,218 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.apache.zeppelin.interpreter.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.ExecuteException;
+import org.apache.commons.exec.ExecuteResultHandler;
+import org.apache.commons.exec.ExecuteWatchdog;
+import org.apache.commons.exec.LogOutputStream;
+import org.apache.commons.exec.PumpStreamHandler;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract class for launching java process.
+ */
+public abstract class ProcessLauncher implements ExecuteResultHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ProcessLauncher.class);
+
+  public enum State {
+    NEW,
+    LAUNCHED,
+    RUNNING,
+    TERMINATED,
+    COMPLETED
+  }
+
+  private CommandLine commandLine;
+  private Map<String, String> envs;
+  private ExecuteWatchdog watchdog;
+  private ProcessLogOutputStream processOutput;
+  protected String errorMessage = null;
+  protected State state = State.NEW;
+  private boolean launchTimeout = false;
+  private final String workingDirectory;
+
+  public ProcessLauncher(CommandLine commandLine,
+                         Map<String, String> envs,
+                         String workingDirectory) {
+    this.commandLine = commandLine;
+    this.envs = envs;
+    this.processOutput = new ProcessLogOutputStream();
+    this.workingDirectory = workingDirectory;
+  }
+
+  public ProcessLauncher(CommandLine commandLine,
+                         Map<String, String> envs,
+                         ProcessLogOutputStream processLogOutput,
+                         String workingDirectory) {
```

Comment:
Pretty much the same `ProcessLauncher` as in Zeppelin, but I've added a working directory here.